hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/statemachine/StateContext.java

```diff
@@ -988,4 +999,39 @@ public DatanodeQueueMetrics getQueueMetrics() {
   public String getThreadNamePrefix() {
     return threadNamePrefix;
   }
+
+  static class PipelineKey {
+    private final HddsProtos.PipelineID pipelineID;
+    private final PipelineAction.Action action;
+
+    PipelineKey(HddsProtos.PipelineID pipelineID, PipelineAction.Action action) {
+      this.pipelineID = pipelineID;
+      this.action = action;
+    }
+
+    public HddsProtos.PipelineID getPipelineID() {
+      return pipelineID;
+    }
+
+    public PipelineAction.Action getAction() {
+      return action;
+    }
```

Comment:
We may have an `equalsId` method instead.
```java
    boolean equalsId(PipelineReport report) {
      return pipelineID.equals(report.getPipelineID());
    }
```

Reply:
Thanks @twosz.
I have updated them all.

Reply:
Thanks @szetszwo .
I have updated them all.